homeassistant/components/homeassistant/triggers/time.py

```diff
@@ -187,10 +222,8 @@ def update_entity_trigger(entity_id: str, new_state: State | None = None) -> Non
                 )
             )
 
-    # Track state changes of any entities.
-    removes.append(
-        async_track_state_change_event(hass, to_track, update_entity_trigger_event)
-    )
+    # Track state changes of all entities
+    removes.extend((async_track_state_change_event(hass, *entry)) for entry in to_track)
```

Comment:
unpacking a tuple like this makes the code difficult to read

Could we use a named tuple instead, and something like:
```
    removes.extend(
        (async_track_state_change_event(hass, entry.entity_id, entry.callback))
        for entry in to_track
    )
```